include/vcpkg/base/stringview.h

```diff
@@ -39,20 +44,23 @@ namespace vcpkg
         constexpr const char* data() const noexcept { return m_ptr; }
         constexpr size_t size() const noexcept { return m_size; }
         constexpr bool empty() const noexcept { return m_size == 0; }
+        constexpr char operator[](size_t pos) const noexcept { return m_ptr[pos]; }
+
+        // intentionally not provided because this may not be null terminated
+        // constexpr const char* c_str() const
 
         std::string to_string() const;
         void to_string(std::string& out) const;
         explicit operator std::string() const { return to_string(); }
 
         StringView substr(size_t pos, size_t count = std::numeric_limits<size_t>::max()) const noexcept;
 
-        constexpr char operator[](size_t pos) const noexcept { return m_ptr[pos]; }
-
     private:
         const char* m_ptr = 0;
         size_t m_size = 0;
     };
 
+    // intentionally not hidden friends
```

Comment:
why are these intentionally not hidden?

Reply:
Because we want

```
StringView sv{"hello"};
sv == "world"
```

to compile.

Reply:
Hmmm come to think of it that may actually work while still hidden friends. I'll just delete the comment.